server/src/infra/configuration.rs

```diff
@@ -81,6 +81,26 @@ impl std::default::Default for LdapsOptions {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, derive_builder::Builder)]
+#[builder(pattern = "owned")]
+pub struct PluginOptions {
+    #[builder(default = "true")]
+    pub enabled: bool,
+    #[builder(default = "false")]
+    pub allow_on_password_update: bool,
+    #[builder(default = r#"None"#)]
+    pub kvscope: Option<String>,
+    pub plugin_path: PathBuf,
+    #[builder(default = r#"None"#)]
+    pub config: Option<BTreeMap<String, String>>,
```

Comment:
An empty map should be enough of a default value, no need to wrap it in an Option. Or does it have a different meaning?

Reply:
I agree, and I have been trying to bend figment to my will, but even with a default like BTreeMap::new(), and similar, I couldn't get it to accept a configuration where that key was missing, if it wasn't option. I know there are others that work fine without Option, by having a default, but I couldn't get it to work for this BTreeMap for some reason.

Reply:
Have a new type wrapping the map, and that has a Default value